### PR TITLE
Added SDK Workflow Parameters and SDK Logging documentation

### PR DESCRIPTION
**Pull request by williamexperian**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1014,6 +1015,24 @@ public StepProcessor createProcessor(final StepProcessorBuilder processorBuilder
 }
 ```
 
+### Workflow parameters
+
+Workflow parameters allow you to configure the settings of your Workflow steps without specifying the value. You can create a Workflow parameter and assign it to any of the Workflow step setting that support the Workflow parameter's datatype.
```
Comment (chungkhenhah) [inline README.md:1020]: should be workflow step property instead of workflow step setting

**Comment by chungkhenhah** [comment]
there is a need to a new new version (2.5.0) to "Compatibility matrix between SDK and Data Studio version", mentioning what is feature being release in 2.5.0

File: README.md (hunk 1 above)
Comment (williamexperian) [reply README.md]: Thank you @chungkhenhah, I have updated the Workflow parameters section :)

File: README.md [hunk 2]
```diff
@@ -1072,6 +1092,20 @@ Some notes on jar packaging:
    ```
 
 ## The Logging library
```
Comment (yudhistira-arya) [inline README.md:1094]: Should mention 2 additional things: 
* That when you `getLogger(some.class)` you can configure it explicitly in the log4j2.xml by defining `<logger>` of the same name. 
* About the fallback mechanism to root logger.

Comment (williamexperian) [reply README.md:1094]: Thank you @yudhistira-arya, I have added the details to the documentation for the SDK Logging section :)

**Pull request merged**